Mathlib/Tactic/Linarith/Frontend.lean

```diff
@@ -389,6 +389,13 @@ syntax (name := nlinarith) "nlinarith" "!"? linarithArgsRest : tactic
 @[inherit_doc nlinarith] macro "nlinarith!" rest:linarithArgsRest : tactic =>
   `(tactic| nlinarith ! $rest:linarithArgsRest)
 
+/-- Elaborate `t` in a way that is suitable for linarith. -/
+def elabLinarithArg (tactic : String) (t : Term) : TacticM Expr := Term.withoutErrToSorry do
+  let (e, mvars) ← elabTermWithHoles t none `linarith (allowNaturalHoles := true)
```

Comment:
```suggestion
def elabLinarithArg (tactic : Name) (t : Term) : TacticM Expr := Term.withoutErrToSorry do
  let (e, mvars) ← elabTermWithHoles t none tactic (allowNaturalHoles := true)
```
This is really just a very minor style suggestion (to be consistent with e.g. `throwTacticEx` and `checkNotAssigned`)—of course, it doesn't matter what we name the mvars. :P

Reply:
Making a good API is hard work, especially with meta code, so I think it's good to get at least one more tactic onboard and factor it out then. That could be a good followup, going through mathlib tactics finding where we're forgetting `Term.withoutErrToSorry` and where we'd never want new metavariables.

Sometimes it's good packaging up common behaviors, but other times the best name for something is its own code (what I mean is, it can be easier to remember and easier to read if the few lines are just repeated every time). This also protects against factoring out accidental temporary alignment of common features, which tends to lead to library code with profusions of options, since as the actually needed features go out of sync, that factored-out common code isn't so common anymore.